### PR TITLE
use yearly ECMAScript naming scheme

### PR DESCRIPTION
**Pull request by michaelficarra**
As discussed in #131.

### CODE REVIEW

**Comment by michaelficarra** [comment]
By the way, I think we should similarly rename the other files to be based on the yearly naming scheme.

**Comment by mikesherov** [comment]
@michaelficarra please do.

**Comment by michaelficarra** [comment]
@mikesherov done

**Comment by hzoo** [comment]
👍  do we want to change the ES7/ES8 reference's in the files in this too?

**Comment by michaelficarra** [comment]
@hzoo done

**Comment by mikesherov** [comment]
:+1: @rreverser merge at will.

**Comment by RReverser** [comment]
As part of this, could we also rename `spec.md` to `es5.md` to avoid confusion? (they're all "specs" after all)

**Comment by mikesherov** [comment]
Makes sense @rreverser.

**Comment by michaelficarra** [comment]
@RReverser done

File: es2015.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
-This document specifies the extensions to the [core ESTree AST types](spec.md) to support the ES6 grammar.
+This document specifies the extensions to the [core ESTree AST types](spec.md) to support the ES2015 grammar.
```
Comment (RReverser) [inline es2015.md]: @michaelficarra I assume this link is now broken?

Comment (michaelficarra) [inline es2015.md]: That's right. A lot of external links will also now be broken. I'll fix the internal links at least.

Comment (michaelficarra) [inline es2015.md]: @RReverser fixed

**Comment by ariya** [comment]
A stop-gap: have a small `spec.md` containing an info that it's now available as `es5.md`.

**Comment by RReverser** [comment]
@ariya That, or even list of all versions with links. Anyway, I don't think many external links pointed directly to `spec.md` but instead to the entire repo, so we should be fine.

**Comment by michaelficarra** [comment]
Okay, should I merge then?

**Comment by mikesherov** [comment]
LGTM

**Comment by michaelficarra** [comment]
FYI it's starting: https://github.com/jquery/esprima/issues/1524

**Pull request merged**